dnsprovider/pkg/dnsprovider/providers/google/clouddns/rrchangeset.go

```diff
@@ -115,7 +119,7 @@ func (c *ResourceRecordChangeset) Apply() error {
 }
 
 func (c *ResourceRecordChangeset) IsEmpty() bool {
-	return len(c.additions) == 0 && len(c.removals) == 0
+	return len(c.additions) == 0 && len(c.removals) == 0 && len(c.upserts) == 0
```

Comment:
Shouldn't there be a unit test for this, covering all of the implementations of the interface method? The only prior caller to this method I could find was a unit test for the case of it being empty. There should be tests for one addition, one removal, and one upsert.

Reply:
Yes, there should be.  We inherited a lot of this code from k/k, and the design is pretty different from the rest of the code (even the rest of the code in k/k).  It's hard to resist just rewriting the whole thing...